src/storage-client/src/storage_collections.rs

```diff
@@ -1524,14 +1525,34 @@ where
                             dependency_since
                         );
 
-                        dependency_since
-                    } else {
-                        data_shard_since
+                            dependency_since
+                        } else {
+                            data_shard_since
+                        }
                     }
-                }
-                None => data_shard_since,
+                    None => data_shard_since,
+                };
+
+                (storage_dependencies, initial_since)
             };
 
+            if since_handle.since().borrow() == AntichainRef::new(&[T::minimum()])
+                && PartialOrder::less_than(
+                    &AntichainRef::new(&[T::minimum()]),
+                    &initial_since.borrow(),
+                )
```

Comment:
In theory this is worth doing whenever the "initial since" is past the handle's since, but I was worried about adding a lot of overhead to large `create_collections` calls. This handles the important case (catching up a brand-new shard to something recent) and shouldn't add noticeable overhead at bootstrap time.

Reply:
Why _do_ we want do downgrade the since here? The controller will use the `initial_since`, so is it about restarts, where we want to make sure that on restart we have all the sinces in consistent state?

Reply:
Discussed this live, but to summarize here:
- Normally the upper is ahead of the since and the since updates don't matter. (As long as they happen eventually.)
- On collection create this is not true: the since advances while the upper is left at 0. The since is the only meaningful frontier we have available.
- Sources are the only place where we rely on frontier info from the shards directly instead of funneling it (directly or indirectly) from the controller. So while most places are fine if we just set the right initial value in the controller, sources care that we flush that out to durable state.